### DISCUSSION TITLE
Accessibility: links do not have proper roles

### DISCUSSION THREAD

**Discussion by dahalsaur** [add2cal/add-to-calendar-button#294] [Ideas / Requests]
![Screenshot 2023-01-13 at 08 57 49](https://user-images.githubusercontent.com/39714407/212268314-82c745ce-daa1-4653-8df0-07323ebd65ea.png)

As you can see, roles of calendar links are generic. In this case, the screen reader won't announce the links "Apple", "Google", "iCal File", etc. with their correct role, which would be confusing to screen reader users. Could you please provide proper role to them?

**Comment by jekuer** [comment]
What would be your proposal and would you add it to the text or the whole `.atcb-list-item`?

**Comment by dahalsaur** [comment]
I think it depends on the type of list item. If its a button, role="button", if its a link, role="link", and so on. https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles

From the first glance, I could see some of the list items triggers file download (like apple), so I would recommend role "button" for those whereas some opens a link in another tab (like google calendar), which I would recommend role "link". 

The main goal is to have a specific role for these list items instead of "generic" role to avoid confusion for screenreader users.

**Reply by jekuer** [reply]
Will add this for the next version.
Thanks.

One more question: What tool/extension/browser did you use in the screenshot?

**Reply by dahalsaur** [reply]
Thanks! I used Google Chrome Version 108.0.5359.124 (Official Build) (arm64) and its native inspect tool. 

By the way, adding a role means programmatically defining the function of the control. So if you use proper html tag (<a>, <button>, etc), you don't have to manually provide roles. I would recommend to use proper html tags if possible.

**Comment by dahalsaur** [comment]
By the way, I have reported another accessibility issues as a separate discussion. Hope that is okay. https://github.com/add2cal/add-to-calendar-button/discussions/295

**Reply by jekuer** [reply]
always!

**Comment by jekuer** [comment]
Should be optimized with the latest V2 version 😃
